### ISSUE TITLE
Infinite loop in prayer trouble handling is possible - Busted calc_total_maxhp 

### ISSUE THREAD

**Issue by demogorgon22** [nethack-cleaner/SlashDNH#2]
TL;DR/ root cause i think? is slashdnh botched calc_total_maxhp  so if polyd and <=5 hp an infinite loop can be entered where the hp trouble isn't ever fixed and continues to be identified.

SlashDNH deleted a closing parenthesis for this else block.
https://github.com/nethack-cleaner/SlashDNH/blob/8378f4c28c0b47f20a0902057e11828254d15970/src/attrib.c#L1175-L1182

Here is what it looks like in DNH. 

https://github.com/Chris-plus-alphanumericgibberish/dNAO/blob/39f764721108f29aba4483b5af0d932c3936f223/src/attrib.c#L1126-L1134

Line 1181 in slashdnh should probably be a closing parenthesis and I think this bug wouldn't be present?

**Comment by demogorgon22** [comment]
https://github.com/nethack-cleaner/SlashDNH/blob/8378f4c28c0b47f20a0902057e11828254d15970/src/attrib.c#L1221-L1222
Closing parenthesis would also need to be removed. Is this a deliberate change or an accident while merging? if it is a deliberate change, figure out how to avoid this loop case with prayer and also fix the indentation probably?
